mm2src/coins/eth.rs

```diff
@@ -887,49 +893,50 @@ async fn withdraw_impl(coin: EthCoin, req: WithdrawRequest) -> WithdrawResult {
 }
 
 #[cfg(feature = "enable-nft-integration")]
+/// `withdraw_erc1155` function returns details of `ERC-1155` transaction including tx hex,
+/// which should be sent to`send_raw_transaction` RPC to broadcast the transaction.
 pub async fn withdraw_erc1155(ctx: MmArc, req: WithdrawErc1155) -> WithdrawNftResult {
-    let ticker = match req.chain {
-        Chain::Bsc => "BNB",
-        Chain::Eth => "ETH",
+    let coin = lp_coinfind_or_err(&ctx, &req.chain.to_ticker()).await?;
+    let (from_addr, to_addr, token_addr, eth_coin) =
+        get_valid_eth_withdraw_addresses(coin, &req.from, &req.to, &req.token_address)?;
+
+    // todo check amount in nft cache, instead of sending new moralis req
```

Comment:
This is something to think about in next PRs. What if the user transfer some of this NFT or all using another wallet, cache would have an outdated amount in such case. Do you have a solution to this in mind?

Reply:
Yep, I definitely need to implement a function, which will check NFT info updates and cache it in db.
There are questions, in what way caching will be working:
(i) checking the updates on background, when program is running (until ctx is stopped).
(ii) checking the updates inside methods like get nft info/history, withdraw nft, when we call them.
(iii) create a separate method for nft updates, so you have to call it yourself using cli / long swipe on mobile etc (less desirable variant, I think)

I would like to discuss db implementation plan with somebody on the 1:1 call. Like we usually did for sprint planning.